# t-тест


In [1]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf
from scipy.stats import t # t-distribution

## sleep equation
Значимость влияния `union`

контрольные переменные `totwrk, age, male`

Неробастный тест

In [3]:
sleep_df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')
mod1 = smf.ols(formula='sleep~union+totwrk+age+male', data=sleep_df).fit()
mod1.summary(slim=True)

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sleep   R-squared:                       0.117
Model:                            OLS   Adj. R-squared:                  0.112
No. Observations:                 706   F-statistic:                     23.14
Covariance Type:            nonrobust   Prob (F-statistic):           5.50e-18
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3461.4144     68.709     50.378      0.000    3326.514    3596.315
union         10.2120     38.236      0.267      0.789     -64.859      85.283
totwrk        -0.1658      0.018     -9.205      0.000      -0.201      -0.130
age            2.7495      1.396      1.970      0.049       0.009       5.490
male          84.7408     34.424      2.462      0.014      17.155     152.327
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.02e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Имеем: $\hat{\beta}_{union}=10.2120$, s.e.($\beta$)=38.236, $t=\frac{\hat{\beta}}{s.e.(\beta)}=0.267$, $P_{union}=0.789$=78.9%

Уровень значимости $\alpha$=5%=0.05

Вывод: $P_{union}>\alpha$ => **коэфициент незначим!**, т.е. гипотеза о том, что `union` не влияет на `sleep`, не отвергается/согласуется с наблюдениями